Similarities checker doesn't ignore decorators with "ignore-signatures" option enabled

### Bug description
Function decorators are not ignored by the similarities checker when the "ignore-signatures" option is enabled.

For me, it makes sense to ignore them as well as parameterized decorators might be applied to many different functions and look almost the same. For example, some frameworks use decorators to make declare API handlers functions like in the following example:

```python
app = FastAPI()

@app.put(
    "/get-something", 
    status_code=200, 
    responses={
        status.HTTP_400_BAD_REQUEST: {
            "model": HTTPExceptionResponseModel,
        },
        status.HTTP_404_NOT_FOUND: {
            "model": HTTPExceptionResponseModel,
        },
        status.HTTP_409_CONFLICT: {
            "model": HTTPExceptionResponseModel,
        },
    },
)
def get_something(task_id: str, response: Response):
    pass
```

It should be easy to fix it, I am going to open PR, but the main question is is it a bug or it was intentionally implemented this way?

### Steps to reproduce

Given multiple files:
```
module\
    __init__.py
    a.py
    b.py
```

`a.py`:
```python
@decorator1(
    arg1=1,
    arg2="2",
    arg3=True,
    arg4=None,
    arg5="",
)
def func_a():
    """Example func."""
```

`b.py`:
```python
@decorator1(
    arg1=1,
    arg2="2",
    arg3=True,
    arg4=None,
    arg5="",
)
def func_b():
    """Example func."""
```


### Command used

```shell
pylint --disable=all --enable=similarities --ignore-signatures=yes module
```


### Pylint output

```shell
************* Module module.b
module/b.py:1:0: R0801: Similar lines in 2 files
==module.a:0
==module.b:0
@decorator1(
    arg1=1,
    arg2="2",
    arg3=True,
    arg4=None,
    arg5="",
) (duplicate-code)

------------------------------------------------------------------
Your code has been rated at 5.00/10 (previous run: 7.50/10, -2.50)
```


### Expected behavior

`duplicate-code` error shouldn't be reported in this case.

### Pylint version

```shell
pylint 2.9.6
astroid 2.6.6
Python 3.8.2 (default, Jun  8 2021, 11:59:35)
```

--- a/CONTRIBUTORS.txt
+++ b/CONTRIBUTORS.txt
@@ -508,6 +508,7 @@ contributors:
 
 * Maksym Humetskyi (mhumetskyi): contributor
   - Fixed ignored empty functions by similarities checker with "ignore-signatures" option enabled
+  - Ignore function decorators signatures as well by similarities checker with "ignore-signatures" option enabled
 
 * Daniel Dorani (doranid): contributor
 

--- a/ChangeLog
+++ b/ChangeLog
@@ -125,6 +125,10 @@ Release date: TBA
   * Limit ``consider-using-tuple`` to be emitted only for in-place defined ``lists``.
 
   * Emit ``consider-using-tuple`` even if list contains a ``starred`` expression.
+  
+* Ignore decorators lines by similarities checker when ignore signatures flag enabled
+
+  Closes #4839
 
 
 What's New in Pylint 2.9.6?

--- a/ChangeLog
+++ b/ChangeLog
@@ -125,7 +125,7 @@ Release date: TBA
   * Limit ``consider-using-tuple`` to be emitted only for in-place defined ``lists``.
 
   * Emit ``consider-using-tuple`` even if list contains a ``starred`` expression.
-  
+
 * Ignore decorators lines by similarities checker when ignore signatures flag enabled
 
   Closes #4839

--- a/pylint/checkers/similar.py
+++ b/pylint/checkers/similar.py
@@ -595,7 +595,7 @@ def stripped_lines(
             chain(
                 *(
                     range(
-                        func.fromlineno,
+                        func.lineno,
                         func.body[0].lineno if func.body else func.tolineno + 1,
                     )
                     for func in functions

--- a/tests/checkers/unittest_similar.py
+++ b/tests/checkers/unittest_similar.py
@@ -171,7 +171,7 @@ def test_ignore_signatures_fail():
         == (
             '''
 9 similar lines in 2 files
-==%s:[1:11]
+==%s:[7:17]
 ==%s:[8:18]
        arg1: int = 3,
        arg2: Class1 = val1,
@@ -183,9 +183,19 @@ def test_ignore_signatures_fail():
 
    def example():
        """Valid function definition with docstring only."""
-TOTAL lines=29 duplicates=9 percent=31.03
+
+6 similar lines in 2 files
+==%s:[0:6]
+==%s:[1:7]
+   @deco1(dval1)
+   @deco2(dval2)
+   @deco3(
+       dval3,
+       dval4
+   )
+TOTAL lines=35 duplicates=15 percent=42.86
 '''
-            % (SIMILAR5, SIMILAR6)
+            % (SIMILAR5, SIMILAR6, SIMILAR5, SIMILAR6)
         ).strip()
     )
 
@@ -198,7 +208,7 @@ def test_ignore_signatures_pass():
     assert (
         output.getvalue().strip()
         == """
-TOTAL lines=29 duplicates=0 percent=0.00
+TOTAL lines=35 duplicates=0 percent=0.00
 """.strip()
     )
 

--- a/tests/input/similar5
+++ b/tests/input/similar5
@@ -1,3 +1,9 @@
+@deco1(dval1)
+@deco2(dval2)
+@deco3(
+    dval3,
+    dval4
+)
 def func1(
     arg1: int = 3,
     arg2: Class1 = val1,

